app/models/pusher.rb

```diff
@@ -116,12 +116,14 @@ def find
     @rubygem.name = name
 
     sha256 = Digest::SHA2.base64digest(body.string)
+    spec_sha256 = Digest::SHA2.base64digest(spec_contents)
 
     @version = @rubygem.versions.new number: spec.version.to_s,
                                      platform: spec.original_platform.to_s,
                                      gem_platform: spec.platform.to_s,
                                      size: size,
                                      sha256: sha256,
+                                     spec_sha256: spec_sha256,
                                      pusher: user,
                                      pusher_api_key: api_key,
                                      cert_chain: spec.cert_chain
```

Comment:
It looks like abbreviate takes off the cert_chain but nothing else used in the Pusher. If you stored it in a var before you serialized the spec maybe you could avoid the extra dup? Not sure if it's worth the extra complexity.

Reply:
yeah i could, it's just a real hassle with the unit tests